plugins/chat/assets/javascripts/discourse/components/chat-composer.hbs

```diff
@@ -1,93 +1,105 @@
-{{#if this.composerService.replyToMsg}}
-  <ChatComposerMessageDetails
-    @message={{this.composerService.replyToMsg}}
-    @icon="reply"
-    @action={{action "cancelReplyTo"}}
-  />
-{{/if}}
+{{! template-lint-disable no-down-event-binding }}
+<div class="chat-composer__wrapper">
+  {{#if (or this.currentMessage.editing this.currentMessage.inReplyTo)}}
+    <ChatComposerMessageDetails
+      @message={{if
+        this.currentMessage.editing
+        this.currentMessage
+        this.currentMessage.inReplyTo
+      }}
+      @icon="pencil-alt"
+      @cancelAction={{this.onCancel}}
+    />
+  {{/if}}
 
-{{#if this.composerService.editingMessage}}
-  <ChatComposerMessageDetails
-    @message={{this.composerService.editingMessage}}
-    @icon="pencil-alt"
-    @action={{this.cancelEditing}}
-  />
-{{/if}}
+  <div
+    role="region"
+    aria-label={{i18n "chat.aria_roles.composer"}}
+    class={{concat-class
+      "chat-composer"
+      (if this.isFocused "chat-composer--focused")
+      (if this.pane.sending "chat-composer--sending")
+      (if
+        this.isSendEnabled
+        "chat-composer--send-enabled"
+        "chat-composer--send-disabled"
+      )
+      (if this.composer.disabled "chat-composer--disabled")
+    }}
+    {{did-update this.didUpdateMessage this.currentMessage.message}}
+    {{did-update this.didUpdateInReplyTo this.currentMessage.inReplyTo}}
+    {{did-insert this.setupAppEvents}}
+    {{will-destroy this.teardownAppEvents}}
+  >
+    <div class="chat-composer__outer-container">
+      <div class="chat-composer__inner-container">
+        <ChatComposerDropdown
+          @buttons={{this.dropdownButtons}}
+          @isDisabled={{this.composer.disabled}}
+          @hasActivePanel={{and
```

Comment:
This is a nitpick, but do you even need the `and` part, can you just do `{{(eq this.chatEmojiPickerManager.picker.context @context)}}`?

Reply:
yes sure